### PR TITLE
3 Bugs 2 Optimizations

### PR DESCRIPTION
**Pull request by marsupial**

### CODE REVIEW

File: lib/UserInterface/textinput/StreamReaderWin.cpp [hunk 1]
```diff
@@ -176,9 +176,20 @@ namespace textinput {
         return false;
       }
     } else {
+      //https://msdn.microsoft.com/en-us/library/windows/desktop/aa365690(v=vs.85).aspx
```
Comment (vgvassilev) [inline lib/UserInterface/textinput/StreamReaderWin.cpp]: Could you explain briefly what is the intent here, instead of pasting only a link.

File: test/SourceCall/unresolved.C [hunk 2]
```diff
@@ -6,7 +6,6 @@
 // LICENSE.TXT for details.
 //------------------------------------------------------------------------------
 
-// RUN: cat %s | %cling
```
Comment (vgvassilev) [inline test/SourceCall/unresolved.C]: In the past we used to have issues when cling crashed, FileCheck could still return success. This is a check against cling crashing.

File: lib/UserInterface/textinput/StreamReaderWin.cpp (hunk 1 above)
Comment (marsupial) [inline lib/UserInterface/textinput/StreamReaderWin.cpp]: The intent is to stop when EOF reached.

Comment (vgvassilev) [inline lib/UserInterface/textinput/StreamReaderWin.cpp]: Is this a change, relevant only for windows? Please add two lines explaining in plain English why we need to do that, instead of a link, which might change in a few years from now.

**Comment by vgvassilev** [comment]
Merged, thanks for your contribution!
